src/_lib/modules/dropdowns/dropdown-menu.directive.ts

```diff
@@ -1,23 +1,43 @@
-import { Directive, HostListener, QueryList, AfterContentInit, ContentChildren, OnDestroy } from '@angular/core';
+import { Directive, HostListener, QueryList, AfterContentInit, ContentChildren, OnDestroy, ElementRef, ChangeDetectorRef } from '@angular/core';
 import { DropdownService } from "./dropdown.service";
 import { DropdownDirective } from "./dropdown.directive";
-import { map } from "rxjs/operators";
+import { filter, map } from "rxjs/operators";
 import { Subscription, Observable, merge } from 'rxjs';
 
 @Directive({
-  selector: '[nwDropdownMenu]',
-  exportAs: 'nw-dropdown-menu'
+    selector: '[nwDropdownMenu]',
+    exportAs: 'nw-dropdown-menu'
 })
 export class DropdownMenuDirective implements AfterContentInit, OnDestroy {
 
     @ContentChildren(DropdownDirective) nestedDropdowns: QueryList<DropdownDirective>;
 
     private _childrenVisibilityToggledSub: Subscription;
+    private _scrollSub: Subscription;
 
-    constructor(private _service: DropdownService) {}
+    constructor(
+        private _service: DropdownService,
+        private _element: ElementRef,
+        private _cdRef: ChangeDetectorRef
+    ) { }
 
     ngAfterContentInit() {
         this._subscribeToChildrenVisbilityToggle();
+        this._subscribeToScrollPosition();
+    }
+
+    private _subscribeToScrollPosition() {
+        this._scrollSub = this._service.toggle$.pipe(
+            filter(opened => opened),
+        ).subscribe(() => {
+            this._cdRef.detectChanges();
+            this._scrollToActiveElement();
+        });
+    }
+
+    _scrollToActiveElement() {
+        const activeElement = this._element.nativeElement.querySelector('.active');
+        activeElement && this._element.nativeElement.scrollTo({ top: activeElement.offsetTop });
```

Comment:
I'd be interested what the others think, but I personally prefer an `if` in these scenarios just because I find this pattern unexpected, and usually have to read it at least twice before I cop what's going on